packages/@eventual/core/src/bucket.ts

```diff
@@ -0,0 +1,218 @@
+import { Readable } from "node:stream";
+import { getBucketHook } from "./internal/bucket-hook.js";
+import { buckets } from "./internal/global.js";
+import {
+  BucketSpec,
+  BucketStreamOptions,
+  BucketStreamSpec,
+  isSourceLocation,
+  SourceLocation,
+} from "./internal/service-spec.js";
+import { DurationSchedule } from "./schedule.js";
+
+export interface BucketObjectReference {
+  key: string;
+  eTag: string;
+  size: number;
+}
```

Comment:
Might be worth cross-referencing these types with CloudFlare's R2 https://developers.cloudflare.com/r2/api/workers/workers-api-reference/#method-specific-types

Reply:
Looks fine, `etag` is `matchIf` in s3 and `onlyIf` in r2. The only difference is r2 doesn't return metadata about objects like size, which can be 0 for cloudflare.